flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/api/TableITCase.scala

```diff
@@ -124,16 +124,22 @@ class TableITCase(tableEnvName: String, isStreaming: Boolean) extends AbstractTe
     assertEquals(ResultKind.SUCCESS_WITH_CONTENT, tableResult.getResultKind)
     val it = tableResult.collect()
     it.close()
-    val jobStatus = try {
-      Some(tableResult.getJobClient.get().getJobStatus.get())
-    } catch {
-      // ignore the exception,
-      // because the MiniCluster maybe already been shut down when getting job status
-      case _: Throwable => None
-    }
-    if (jobStatus.isDefined) {
-      assertNotEquals(JobStatus.RUNNING, jobStatus.get)
+
+    val numRetries = 7
```

Comment:
Why use retries when you could wait for the job to terminate via `ClusterClient#requestJobResult()`?

Reply:
I'm not sure what will happen if the job never terminates. Will CI stuck forever or does it have an internal time limit? If there is no negative impact on CI I agree we should wait for the job to terminate.

Reply:
I guess, @zentol suggested (correct me if I'm wrong here) something like the following instead of a loop:
```
    val jobId = tableResult.getJobClient.get().getJobID
    AbstractTestBase.MINI_CLUSTER_RESOURCE.getClusterClient.requestJobResult(jobId).get()
```